Lib/test/test_tools/test_i18n.py

```diff
@@ -517,6 +517,219 @@ def test_parse_keyword_spec(self):
                 self.assertEqual(str(cm.exception), message)
 
 
+class Test_msgfmt(unittest.TestCase):
+    """Tests for the msgfmt.py tool
+        bpo-35335 - bpo-9741
+    """
+
+    script = os.path.join(toolsdir,'i18n', 'msgfmt.py')
+
+    # binary images of tiny po files
+    # windows end of lines for first one
+    file1_fr_po = b'''# French translations for python package.\r
+# Copyright (C) 2018 THE python\'S COPYRIGHT HOLDER\r
+# This file is distributed under the same license as the python package.\r
+# s-ball <s-ball@laposte.net>, 2018.\r
+#\r
+msgid ""\r
+msgstr ""\r
+"Project-Id-Version: python 3.8\\n"\r
+"Report-Msgid-Bugs-To: \\n"\r
+"POT-Creation-Date: 2018-11-30 23:46+0100\\n"\r
+"PO-Revision-Date: 2018-11-30 23:47+0100\\n"\r
+"Last-Translator: s-ball <s-ball@laposte.net>\\n"\r
+"Language-Team: French\\n"\r\n"Language: fr\\n"\r
+"MIME-Version: 1.0\\n"\r
+"Content-Type: text/plain; charset=UTF-8\\n"\r
+"Content-Transfer-Encoding: 8bit\\n"\r
+"Plural-Forms: nplurals=2; plural=(n > 1);\\n"\r
+\r
+#: file1.py:6\r
+msgid "Hello!"\r
+msgstr "Bonjour !"\r
+\r
+#: file1.py:7\r
+#, python-brace-format\r
+msgid "{n} horse"\r
+msgid_plural "{n} horses"\r
+msgstr[0] "{n} cheval"\r
+msgstr[1] "{n} chevaux"\r
+'''
+
+    # Unix end of file and a non ascii character for second one
+    file2_fr_po = rb'''# French translations for python package.
+# Copyright (C) 2018 THE python'S COPYRIGHT HOLDER
+# This file is distributed under the same license as the python package.
+# s-ball <s-ball@laposte.net>, 2018.
+#
+msgid ""
+msgstr ""
+"Project-Id-Version: python 3.8\n"
+"Report-Msgid-Bugs-To: \n"
+"POT-Creation-Date: 2018-11-30 23:57+0100\n"
+"PO-Revision-Date: 2018-11-30 23:57+0100\n"
+"Last-Translator: s-ball <s-ball@laposte.net>\n"
+"Language-Team: French\n"
+"Language: fr\n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+"Plural-Forms: nplurals=2; plural=(n > 1);\n"
+
+#: file2.py:6
+msgid "It's over."
+msgstr "C'est termin\xc3\xa9."
+
+#: file2.py:7
+msgid "Bye..."
+msgstr "Au revoir ..."
+'''
+
+    # binary images of corresponding compiled mo files
+    file1_fr_mo = b'\xde\x12\x04\x95\x00\x00\x00\x00\x03\x00\x00\x00\x1c' \
+                  b'\x00\x00\x004\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' \
```

Comment:
Not very important: escaping newlines with backslashes is not recommended, better use parens. This also gives more room per line:
```python
file1_fr_mo = (
   b'bytes bytes bytes bytes etc to 80 columns'
)
```